lib/bpmachine/process_specification.rb

```diff
@@ -1,15 +1,13 @@
 module BPMachine
   module ProcessSpecification
-    def self.after_processes(&block)
-      after_process_actions << block
-    end
-
-    def self.after_process_actions
-      @after_process_actions ||= []
-    end
-
     def self.included(klass)
       klass.extend ClassMethods
+
+      klass.class_eval do
+        def self.included(klass)
+          klass.instance_variable_set :@bpm_module, self
```

Comment:
O problema é suportar o spec "should support modules" e todos os demais. Se alguém tiver uma sugestão, beleza, pois aqui deu timeout.